src/reducers/nft/index.js

```diff
@@ -30,6 +31,13 @@ const nftSlice = createSlice({
                 const { contractName, tokens, accountId } = payload;
                 set(state, ['ownedTokens', 'byAccountId', accountId, 'byContractName', contractName], tokens);
             },
+            addTokensMetadata(state, { payload }) {
+                const { contractName, tokens, accountId } = payload;
+                update(state, ['ownedTokens', 'byAccountId', accountId, 'byContractName', contractName], (n) => ([
+                    ...n,
```

Comment:
1. I would propose something like that instead of condition and `set`:
`update(state, ['ownedTokens', 'byAccountId', accountId, 'byContractName', contractName], (n) => (n || []).concat(tokens));`
2. It's there:
https://github.com/near/near-wallet/pull/1986/commits/9bddb63f319504cdb0bf16d3b058070ca9b4511a

Reply:
:+1: cool I like it! I'm not too fussed on the exact syntax as long as we're taking advantage of immer.js where we can :)